applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_structural_response_function.cpp

```diff
@@ -0,0 +1,942 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder 
+//   
+
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "adjoint_structural_response_function.h"
+
+
+// Application includes
+
+namespace Kratos
+{
+
+    /// Constructor.
+    AdjointStructuralResponseFunction::AdjointStructuralResponseFunction(ModelPart& rModelPart, Parameters& rParameters)
+      : mrModelPart(rModelPart)
+    {
+        KRATOS_TRY;
+
+        mSensitivityModelPartName =
+           rParameters["sensitivity_model_part_name"].GetString();
+
+        Parameters nodal_sensitivity_variables = rParameters["nodal_sensitivity_variables"];
+        mNodalSensitivityVariables.resize(nodal_sensitivity_variables.size());
+        for (unsigned int i = 0; i < nodal_sensitivity_variables.size(); ++i)
```

Comment:
There seem to be a lot of these for loops which fill a `std::vector<std::string` from parameters. You can create a function to do this once and call it in all the other places.

Reply:
You can also return the vector from the function instead of passing it as a parameter since the move operations will make this efficient.